tfe/resource_tfe_project_policy_set.go

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package tfe
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceTFEProjectPolicySet() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceTFEProjectPolicySetCreate,
+		Read:   resourceTFEProjectPolicySetRead,
+		Delete: resourceTFEProjectPolicySetDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceTFEProjectPolicySetImporter,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"policy_set_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceTFEProjectPolicySetCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(ConfiguredClient)
+
+	policySetID := d.Get("policy_set_id").(string)
+	projectID := d.Get("project_id").(string)
+
+	policySetAddProjectsOptions := tfe.PolicySetAddProjectsOptions{}
+	policySetAddProjectsOptions.Projects = append(policySetAddProjectsOptions.Projects, &tfe.Project{ID: projectID})
```

Comment:
Do we anticipate the `PolicySetAddProjectsOptions` struct to grow in the future? It currently only has one field. I noticed you are referencing the `Projects` field directly. I ask because if the struct were to grow, creating an options struct like [this](https://github.com/hashicorp/terraform-provider-tfe/blob/main/tfe/resource_tfe_notification_configuration.go#L288-L293) one could be an alt.

Reply:
@ker-an do you think there will be additional attributes for Options in the future?

Reply:
We are modelling this after [policySetAddWorkspacesOptions](https://github.com/hashicorp/terraform-provider-tfe/blob/03776af85cd657da8b8afb1dfbfb594ff064f7b5/tfe/resource_tfe_workspace_policy_set.go#L49) and it looks like no, we won't be adding attributes in the future.

Reply:
Ok, just curious. Not a blocker. Marking this as resolved. Thanks @ker-an and @Netra2104